stl/src/filesystem.cpp

```diff
@@ -854,21 +824,32 @@ struct alignas(long long) _Aligned_file_attrs {
             _Flags, _Get_file_attributes_data)) { // caller wants something GetFileAttributesExW might provide
         if (_Symlink_attribute_hint == __std_fs_file_attr::_Invalid
             || !_Bitmask_includes(_Symlink_attribute_hint, __std_fs_file_attr::_Reparse_point)
-            || !_Follow_symlinks) { // we might not be a symlink or not following symlinks, so GetFileAttributesExW
+            || !_Follow_symlinks) { // we might not be a symlink or not following symlinks, so FindFirstFileW
                                     // would return the right answer
-            _Aligned_file_attrs _Aligned_attrs;
-            auto& _Data = _Aligned_attrs._Data;
-            if (!GetFileAttributesExW(_Path, GetFileExInfoStandard, &_Data)) {
-                return __std_win_error{GetLastError()};
+
+            // check for file names that contain `?` or `*` (globbing characters)
+            // these are invalid file names, and will give us the wrong answer with `FindFirstFile`
+            if (::wcspbrk(_Path, L"?*")) {
+                return __std_win_error(ERROR_INVALID_NAME);
             }
 
+            WIN32_FIND_DATAW _Data;
+            HANDLE _Find_handle = FindFirstFileW(_Path, &_Data);

```

Comment:
I'm nervous about doing this to `__std_fs_get_stats` (as opposed to just `exists`) because the values returned from `FindFirstFile` are not necessarily up to date, which matters for file size and last write time values. See https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-findfirstfilew#remarks

Reply:
We can't do this for just `exists` because `exists` is defined to be exactly `exists(status(p))`

Reply:
Random idea: try GetFileAttributes and if it doesn't work, call FindFirstFile? What about opening a handle with dwDesiredAccess of 0 and then calling GetFileAttributesByHandle?

Reply:
I believe that the "filesystem data races" wording allows us to ignore this issue if we so choose.

Reply:
> We can't do this for just `exists` because `exists` is defined to be exactly `exists(status(p))`

Yes, which is why I was saying "unfortunate but I think the current implementation does the right thing"